uast/uast.go

```diff
@@ -262,87 +270,311 @@ type Any interface{}
+//
+// Similar to import in PHP, Python, JS.
+//
+// For other import types, see Import.
 type RuntimeImport Import
 
+// RuntimeReImport is a subset of RuntimeImport statement that will re-execute an import and its
+// side-effects statement each time an execution reaches the statement.
+//
+// Similar to import in PHP, Python.
+//
+// For other import types, see Import.
 type RuntimeReImport RuntimeImport
 
+// InlineImport is a subset of import statement that acts like a preprocessor - all statements in
+// the imported module are copied into a position of the UAST node.
+//
+// Similar to import in C and C++.
+//
+// For other import types, see Import.
 type InlineImport Import
 
+// Argument is a named argument or return of a function.
 type Argument struct {
 	GenNode
-	Name        *Identifier `json:"Name"`
-	Type        Any         `json:"Type"`
-	Init        Any         `json:"Init"`
-	Variadic    bool        `json:"Variadic"`
-	MapVariadic bool        `json:"MapVariadic"`
-	Receiver    bool        `json:"Receiver"`
+	// Name is an optional name of an argument.
+	Name *Identifier `json:"Name"`
+
+	// Type is an optional type of an argument.
+	Type Any `json:"Type"`
+
+	// Init is an expression that should be used to initialize an argument
+	// in case no value is provided. Also known as a default value of an argument.
+	Init Any `json:"Init"`
+
+	// Variadic flag may be set only for the last argument to denote a function
+	// with a variadic number of arguments.
+	Variadic bool `json:"Variadic"`
+
+	// MapVariadic flag may be set only for the last argument to denote a function
+	// that accepts a map/dictionary value that is mapped to function arguments.
+	MapVariadic bool `json:"MapVariadic"`
+
+	// Receiver is set to true if an argument is a receiver of a method call.
+	Receiver bool `json:"Receiver"`
 }
 
+// FunctionType is a signature of a function.
 type FunctionType struct {
 	GenNode
+	// Arguments is a set of arguments the function accepts.
+	//
+	// Methods defined on structures and classes should have the first argument
+	// that corresponds to a method's receiver ("this" in most languages).
 	Arguments []Argument `json:"Arguments"`
-	Returns   []Argument `json:"Returns"`
+
+	// Returns is a set of values returned by a function.
+	//
+	// Languages with an implicit return should specify a single return with an
+	// unspecified type.
+	Returns []Argument `json:"Returns"`
 }
 
+// Function is a declaration of a function with a specific signature and implementation.
+//
+// Name is not a part of function declaration. Use Alias as a parent node to specify
+// the name of a function.
+//
+// What is considered a Function:
+// - function declaration;
+// - anonymous functions;
 type Function struct {
 	GenNode
+	// Type is a signature of a function. Should always be set.
 	Type FunctionType `json:"Type"`
-	Body *Block       `json:"Body"`
+
+	// Body is an optional implementation of a function. should point to a Block with
```

Comment:
Why is this optional? If the declaration has no body, shouldn't it be a `FunctionType`?

Reply:
I assumed forward declaration may clarify as `Function` as well as a function header that is implemented in another language (Go function with WASM implementation, Java and JNI).